fetch-in-react-practice/client/components/app.jsx

```diff
@@ -0,0 +1,86 @@
+import React from 'react';
+import PageTitle from './page-title';
+import TodoList from './todo-list';
+import TodoForm from './todo-form';
+
+export default class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      todos: []
+    };
+    this.addTodo = this.addTodo.bind(this);
+    this.toggleCompleted = this.toggleCompleted.bind(this);
+  }
+
+  componentDidMount() {
+    this.getAllTodos();
+  }
+
+  getAllTodos() {
+    fetch('/api/todos')
+      .then(res => res.json())
+      .then(users => this.setState({
+        todos: users
+      }));
+  }
+
+  addTodo(newTodo) {
+    fetch('/api/todos', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify(newTodo)
+    })
+      .then(response => response.json())
+      .then(data => {
+        this.setState({ todos: data });
+      })
+      .catch(error => {
+        console.error('Error:', error);
+      });
+  }
+
+  toggleCompleted(todoId) {
+    const todos = this.state.todos;
+    let completedIndex = 0;
+    todos.forEach((obj, index, array) => {
+      if (obj.todoId === todoId) {
+        completedIndex = index;
+      }
+    });
```

Comment:
This definitely works, but there are built in array methods for doing this. The best one to use here would be [Array.prototype.findIndex()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findIndex)